### PR TITLE
multiple idps and identity transformations docs

### PR DESCRIPTION
**Pull request by cfryanr**
Website doc changes to go along with #1419.

**Release note**:

None, docs only.

```release-note
NONE
```

### CODE REVIEW

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:8]: How about:
- `Multiple IDPs`
- `Weight 100`

Simple link, last in the list, after all of the `With <single IDP>` so its fairly obvious that its a different config?

Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:8]: The new doc is about how to configure any IDPs going forward... one or many, not just how to configure multiple. We've been calling the feature "multiple IDPs" as a shorthand, but really it is much bigger than that. I'd rather avoid naming it anything about "multiple".

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:257]: Noting quick that we saw the comments wrap so we should shorten the lines:

Screenshot 2023-09-12 at 1 33 18 PM

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:9]: I think we should weight this menu item last.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (joshuatcasey) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:115]: "usernames FROM one IDP"

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:8]: Screenshot 2023-09-12 at 1 37 15 PM

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (joshuatcasey) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:159]: "leave the GROUPS unchanged".

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (joshuatcasey) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:161]: "Returning true HAS no"

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (joshuatcasey) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:166]: perhaps "which uses this FederationDomain"?

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (joshuatcasey) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:170]: "All three transformation expressions"?

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:8]: - `Required Config`
- `FederationDomain Config`
?

**Review by joshuatcasey — APPROVED** [review]

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:13]: Elsewhere we link [FederationDomain](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.27/README.adoc#federationdomain) to this doc.  I think we should do that here as well.  Long term, I think we should add a `Custom Resources` menu item and detail out what each one is.  `FederationDomain` is an important concept, but as I read our docs I get the sense that they are written assuming people understand the concept.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:22]: Not sure if we want to link  link [FederationDomain](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.27/README.adoc#federationdomain)  every time?  Maybe just the first time in the document is best.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:26]: I think Josh and I talked about this, I tend to want to see our custom resources always in the mono font in our docs, like `OIDCIdentityProvider`, `ActiveDirectoryIdentityProvider`, `LDAPIdentityProvider`.  Prob a conversation to spin out later, but this is a good docs PR to trigger the conversation.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:20]: I do find [configure the Supervisor as an OIDC issuer](https://deploy-preview-1660--pinniped.netlify.app/docs/howto/supervisor/configure-supervisor/) a little disorienting as it links to `Required Configuration` in the menu.  I think either we should align the text with some kind of symmetry or say:

```
 read the guide about how to configure the Supervisor as an OIDC issuer via the [Required Configuration document](#)
```

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:31]: nit: I tend to want to lead this sentence with `If there is exactly one IdentityProvider configured in the supervisor namespace, and there is no spec.identityProviders...` just to put the key point of backwards identity first.

(nit: meaning, not a big deal, feel free to ignore/resolve).

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:26]: For this PR, feel free to resolve.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:46]: I think we should indicate that the first way is `depcrecated` or may be deprecated in the future and the second way is the preferred/blessed way.  Getting people on board now will make deprecation easier in the future, should we choose to do so.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:47]: I wonder if we want to point out the `IdentityProvidersFound` status here. 🤔

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:53]: Should we link to `docs/howto/install-supervisor/`?

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:93]: This is a great section.  In the future I'm a bit inclined to pull it out into a separate page, with a link in the menu, perhaps under a heading `Security Features`.  I think if we can bubble up the interesting security features Pinniped provides and make them easy to find it will benefit the project.

I'd add things like `kubeconfig generation` to this sub-menu as well, discussing how credentials are not stored in the kubeconfig, how it can (usually) be shared safely as it contains minimal information, etc.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:108]: Yeah, in the future breaking this down into several pages so we can add a menu item for `disambiguation` as an easy to find feature would be nice.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:128]: Definitely would be a great page to break out and add as a security feature menu item.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:135]: I'd love to link to a `FederationDomain` custom resource page with a section about this, rather than just make it bold.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:141]: Shall we link to Kubernetes CEL docs? 
https://kubernetes.io/docs/reference/using-api/cel/.  Since CEL is relatively new we may want to emphasize how these are becoming standard in Kubernetes and not an arbitrary decision made by Pinniped.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:152]: The `bold` here makes me want to click a link and see the whole resource.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:263]: Awesome detailed example.  Just need to tweak the comment wrapping.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (benjaminapetersen) [inline site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:349]: Super helpful examples.

**Review by benjaminapetersen — APPROVED** [review]
Most of my comments are tiny `nit` or "I'd love to break this up to make it easier for users to find specific features!", so nothing to really hold this PR back from merge once the feature merges.  

🥳 🎉 🥳 🎉 🥳 🎉 🥳

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:263]: Done

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:20]: Let's consider reorganizing this new page and the pre-existing "Required Configuration" page in a future PR. For this PR, I just want to get the information about how to use the new features onto the site.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:31]: It is first to be symmetrical with the leading sentence of the second bullet point. The difference between the two cases are whether or not you explicitly specify `spec.identityProviders`.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:13]: Updated.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:22]: Agree that the first time is enough.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:46]: Agreed except we need to implement an interstitial page for web app clients or else they will not work using the new way. After that feature is done, then we should go back to this doc and add some text about how the old way will be deprecated.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:47]: Not sure what you mean here.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:8]: Let's consider reorganizing this new page and the pre-existing "Required Configuration" page in a future PR. For this PR, I just want to get the information about how to use the new features onto the site.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:93]: Let's consider reorganizing this new page and the pre-existing "Required Configuration" page in a future PR. For this PR, I just want to get the information about how to use the new features onto the site.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:108]: Let's consider reorganizing this new page and the pre-existing "Required Configuration" page in a future PR. For this PR, I just want to get the information about how to use the new features onto the site.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:128]: Let's consider reorganizing this new page and the pre-existing "Required Configuration" page in a future PR. For this PR, I just want to get the information about how to use the new features onto the site.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:135]: Updated to link the first time (not every time).

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:141]: I don't think we need to defend our choices in the user guide. We did that in the proposal doc.

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:53]: Updated

File: site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md
Comment (cfryanr) [reply site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md:152]: Updated

**Review by joshuatcasey — APPROVED** [review]

**Pull request merged**
